### PR TITLE
fix: fix all mutated props

### PR DESCRIPTION
**Pull request by asbiin**
Remove all mutated props in Vue pages.

You fool, don't forget these steps:

- [x] Unit tests
- [x] Tests with Cypress
- [ ] Documentation
- [ ] Dummy data

### CODE REVIEW

File: resources/js/Pages/Adminland/CompanyNews/Index.vue [hunk 1]
```diff
@@ -34,20 +34,22 @@
       <div class="mw7 center br3 mb5 bg-white box restricted relative z-1">
         <div class="pa3 mt5">
           <h2 class="tc normal mb4">
-            {{ $t('account.company_news_title', { company: $page.props.auth.company.name}) }}
+            {{ $t('account.companylocalNews_title', { company: $page.props.auth.company.name}) }}
```
Comment (djaiss) [inline resources/js/Pages/Adminland/CompanyNews/Index.vue:37]: No no 😀 you’ve also changed the translation strings, which won’t work, throughout the entire file and others too.

**Review by djaiss — CHANGES_REQUESTED** [review]
Lots of i18n strings have been changed, which is not right.

File: resources/js/Pages/Adminland/CompanyNews/Index.vue (hunk 1 above)
Comment (asbiin) [reply resources/js/Pages/Adminland/CompanyNews/Index.vue:37]: Good catch! thanks

File: resources/js/Pages/Adminland/Expense/Partials/Categories.vue [hunk 2]
```diff
@@ -145,6 +146,16 @@ export default {
     };
   },
 
+  watch: {
+    categories(value) {
```
Comment (djaiss) [inline resources/js/Pages/Adminland/Expense/Partials/Categories.vue:150]: why do you do this? i don’t see the need

**Review by djaiss — COMMENTED** [review]
Have you tested each ones of the view with our Cypress tests to see if something is broken? I don’t think it would fail, but still.

File: resources/js/Pages/Adminland/Expense/Partials/Categories.vue (hunk 2 above)
Comment (asbiin) [reply resources/js/Pages/Adminland/Expense/Partials/Categories.vue:150]: When you use the template in a page, you'll bind the value: `<categories :categories="valueFromPage" />`.
In the Categories component, the value is copied in `localCategories` to be used and changed. But if the `valueFromPage' change inside the Page after everything is mounted, you will have to update the `localCategories`. The `watch` function do that.

**Comment by asbiin** [comment]
100% Cypress tests OK!

File: resources/js/Shared/Select.vue [hunk 3]
```diff
@@ -28,7 +28,7 @@
               :label="customLabelKey"
               :data-cy="datacy"
               :close-on-select="true"
-              @input="broadcast(selected)"
+              @input="broadcast"
```
Comment (djaiss) [inline resources/js/Shared/Select.vue:31]: are you sure here? how does it know what's selected?

**Review by djaiss — COMMENTED** [review]
I just checked the code itself - I haven't run the tests or Cypress to see if it still works.

Good job!

One thing to validate though - see comment.

File: resources/js/Shared/Select.vue (hunk 3 above)
Comment (asbiin) [reply resources/js/Shared/Select.vue:31]: The value is sent in the event's parameter, to the `broadcast` function.

Comment (asbiin) [reply resources/js/Shared/Select.vue:31]: See https://vue-select.org/guide/values.html#getting-and-setting

**Review by djaiss — APPROVED** [review]

**Pull request merged**